src/providers/mbed_provider/mod.rs

```diff
@@ -105,13 +104,13 @@ fn create_key_id(
     key_triple: KeyTriple,
     store_handle: &mut dyn ManageKeyIDs,
     local_ids_handle: &mut LocalIdStore,
-) -> Result<psa_crypto_binding::psa_key_id_t> {
-    let mut key_id = rand::random::<psa_crypto_binding::psa_key_id_t>();
+) -> Result<KeyId> {
+    let mut key_id = rand::random::<KeyId>();
     while local_ids_handle.contains(&key_id)
-        && key_id != 0
-        && key_id < constants::PSA_MAX_PERSISTENT_KEY_IDENTIFIER
+        || key_id == 0
+        || key_id > constants::PSA_MAX_PERSISTENT_KEY_IDENTIFIER
```

Comment:
Good catch :P

Reply:
I'm still horrified I made that mistake in the first place.

Reply:
It happens to the best of us.